llama_index/download/dataset.py

```diff
@@ -0,0 +1,243 @@
+"""Download."""
+
+import json
+import os
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import requests
+import tqdm
+
+from llama_index.download.utils import (
+    initialize_directory,
+)
+
+LLAMA_DATASETS_CONTENTS_URL = (
+    f"https://raw.githubusercontent.com/run-llama/llama_datasets/main"
+)
+LLAMA_DATASETS_PATH = "/llama_datasets"
+LLAMA_DATASETS_URL = LLAMA_DATASETS_CONTENTS_URL + LLAMA_DATASETS_PATH
+
+LLAMA_DATASETS_LFS_URL = (
+    f"https://media.githubusercontent.com/media/run-llama/llama_datasets/main"
+)
+
+LLAMA_DATASETS_SOURCE_FILES_GITHUB_TREE_URL = (
+    "https://github.com/run-llama/llama_datasets/tree/main"
+)
+LLAMA_RAG_DATASET_FILENAME = "rag_dataset.json"
+LLAMA_SOURCE_FILES_PATH = "source_files"
+
+
+PATH_TYPE = Union[str, Path]
+
+
+def _get_file_content(loader_hub_url: str, path: str) -> Tuple[str, int]:
+    """Get the content of a file from the GitHub REST API."""
+    resp = requests.get(loader_hub_url + path)
+    return resp.text, resp.status_code
+
+
+def _get_file_content_bytes(loader_hub_url: str, path: str) -> Tuple[bytes, int]:
+    """Get the content of a file from the GitHub REST API."""
+    resp = requests.get(loader_hub_url + path)
+    return resp.content, resp.status_code
+
+
+def _get_source_files_list(source_tree_url: str, path: str) -> List[str]:
+    """Get the list of source files to download."""
+    resp = requests.get(source_tree_url + path + "?recursive=1")
+    payload = resp.json()["payload"]
+    return [item["name"] for item in payload["tree"]["items"]]
+
+
+def get_dataset_info(
```

Comment:
isn't this the same function as `get_module_info`

Reply:
not quite, it was before I added logic for listing the files in `llama_datasets/<dataset_id>/source_files`. The `get_module_info` doesn't need to do this. As a result, in the datasets `library.json`, we don't need to have really long list/entries for `extra_files` field — in fact, for datasets, it would no longer be needed.